plugins/pulp_rpm/plugins/distributors/yum/publish.py

```diff
@@ -834,66 +835,62 @@ def process_main(self, item=None):
 class PublishModulesStep(platform_steps.UnitModelPluginStep):
     """
     Publish modularity artifacts.
-
-    :ivar fp_out: file-like used for writing the modules.yaml.
-    :type fp_out: file-like
     """
 
-    FILE_NAME = 'modules.yaml.gz'
-
     def __init__(self):
         super(PublishModulesStep, self).__init__(
             constants.PUBLISH_MODULES_STEP,
             [
                 models.Modulemd,
                 models.ModulemdDefaults
             ])
+        self.context = None
         self.description = _('Publishing Modules')
-        self.fp_out = None
+
+    def is_skipped(self):
+        """
+        Test to find out if the step should be skipped.
+
+        :return: whether or not the step should be skipped
+        :rtype:  bool
+        """
+        # skip if there are no modules files.
+        if self.get_total() == 0:
+            return True
+
+        return super(PublishModulesStep, self).is_skipped()
 
     def initialize(self):
         """
         Create the directory and open fp_out for writing.
         """
-        path = os.path.join(
-            self.get_working_dir(),
-            REPO_DATA_DIR_NAME,
-            self.FILE_NAME)
-        plugin_misc.mkdir(os.path.dirname(path))
-        self.fp_out = gzip.open(path, 'wb')
+        checksum_type = self.parent.get_checksum_type()
+        self.context = ModulesFileContext(self.get_working_dir(), checksum_type=checksum_type)
+        self.context.initialize()
 
     def process_main(self, item=None):
         """
         Append each yaml document to the modules.yaml.
 
         :param item: Either Modulemd or ModulemdDefaults.
-        :type item: pulp.server.db.model.FileContent
+        :type item: pulp.server.db.model.FileContent (baseclass)
         """
         with open(item.storage_path) as fp_in:
             document = fp_in.read()
-            checksum = hashlib.sha256(document).hexdigest()
```

Comment:
I'm not 100% sure why this check is necessary, but given that it was considered important enough to do + assign a coded exception to, I'm leaving it here.

Reply:
@dalley, is it still a valid comment? or an outdated one? I don't follow what you mean here. It looks like the code which performs a check is removed

Reply:
Ah I see, you moved it to `metadata/modules.py`. Never mind.